### DISCUSSION TITLE
Maintain Public IP for Codespaces container

### DISCUSSION THREAD

**Discussion by MSFTJim** [community/community#5677] [Codespaces]
Is it possible to maintain the public IP address for a Codespace or maybe have a range of dynamic IPs? 

Dev team using them to whitelist in firewall to access remote services.

**Comment by tanmayeekamath** [comment]
Hey @MSFTJim, thanks for your feedback. The current IP address range for Codespaces is dynamic. However, we do have it in our backlog to support a fixed IP range for codespaces to enable allowlisting those to be accessible from your internal services or vice versa. I'll update this thread once we have a timeline on when this would be supported. 

Can you share a bit more on where your services are (ex. VPNs, cloud based resources etc.) and what does your networking set up look like?

**Comment by MSFTJim** [comment]
Hello Tanmayee,

I was working with a Partner who at various stages in their development cycle for clients, create replicas of the clients on-prem environment in virtual networks running in Azure.  Sometimes they are just vm's other times they could include Azure services.  They typically just whitelist the IP addresses of their team to access the vnet.

For the developer desktops, I suggested they consider Codespaces (I am a big fan of Codespaces!).  We tested it out but discovered our Codespaces had a different IP each time they restarted the Codespace.  Updating the nsg that frequently just proved to be more effort so they switched back to local desktop environments.

Let me know if you want anymore detail.  Thanks.
